### ISSUE TITLE
[QUESTION] Is there a way to access the model's history of training and validation loss values?

### ISSUE THREAD

**Issue by giacomoguiduzzi** [unit8co/darts#2565]
Greetings,

I searched the documentation but couldn't find anything regarding the argument. I may have lost it so please point me in the right direction if that's the case. 
After evaluating the model against the test set, I wanted to save the model's losses obtained during training and validation for the various epochs. However, I couldn't find any attribute that saves them regarding Darts models. Are they saved somewhere? Can we access them through the PyTorch Lightning implementation?
Thanks in advance, I'm looking forward to your kind response.

Best Regards,
Giacomo Guiduzzi

**Comment by dennisbader** [comment]
Yes, you can use a custom callback for that:

```
import pytorch_lightning as pl
from pytorch_lightning.callbacks import Callback

from darts.models import TiDEModel

class StepLossLogger(Callback):
    """StepLossLogger is callback to monitor the training and validation losses of torch models
    over the course of the training process on a step/batch level."""

    def __init__(self):
        self.train_step = []
        self.train_loss = []
        self.val_step = []
        self.val_loss = []

    def on_train_batch_end(self, trainer: pl.Trainer, *args, **kwargs) -> None:
        self.train_step.append(int(trainer.global_step))
        self.train_loss.append(float(trainer.callback_metrics["train_loss"]))

    def on_validation_end(self, trainer: pl.Trainer, *args, **kwargs) -> None:
        self.val_step.append(int(trainer.global_step))
        self.val_loss.append(float(trainer.callback_metrics["val_loss"]))


loss_logger = StepLossLogger()
model = TiDEModel(..., pl_trainer_kwargs={"callbacks": [loss_logger]})
```

**Comment by giacomoguiduzzi** [comment]
Hi Dennis,

Thank you so much for pointing that out. I had the feeling it could have to do with PyTorch Lightning. I'm going to test it and close the issue as soon as I can make it work.

Have a great day ahead,
Giacomo Guiduzzi

**Comment by giacomoguiduzzi** [comment]
Hi,

I was able to retrieve the losses. I'm going to report here the way I did it in case it might be of help to someone, as I exploited the Trainer instance.
1. First of all, I created a callback instance: `loss_logger = StepLossLogger()`
2. Then I created a list of callbacks in input to the Trainer constructor:
  ```
  callbacks = [early_stopping, loss_logger]
  trainer = Trainer(
        devices=devices_indexes,
        callbacks=callbacks,
        max_epochs=model_n_epochs,
  )
  ```
3. I called my wrapper class' train method, passing the trainer to the `fit_from_dataset` method:
  ```
  self.model.fit_from_dataset(
        train_dataset=self.train_data,
        val_dataset=self.val_data,
        trainer=trainer,
        verbose=True,
  )
  ```
4. After training, I retrieved the available callbacks from the Trainer object and saved the losses:
  ```
  for callback in trainer.callbacks:
      if isinstance(callback, StepLossLogger):
          self.losses["train"] = callback.train_loss
          self.losses["val"] = callback.val_loss
          break
  ```

Thank you once again for your assistence!

Best Regards,
Giacomo Guiduzzi
